### PR TITLE
Make GetProfileById authorized on profile ownership

### PR DESCRIPTION
**Pull request by jhrozek**
We didn't set any permissions on this method which always denied access.

Fixes: #1935

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]

File: pkg/api/protobuf/go/minder/v1/minder.pb.go [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.31.0-devel
+// 	protoc-gen-go v1.31.0
```
Comment (jhrozek) [inline pkg/api/protobuf/go/minder/v1/minder.pb.go:18]: there seems to be a lot of spam in this diff is it related to the different protoc version I have?

Comment (JAORMX) [reply pkg/api/protobuf/go/minder/v1/minder.pb.go:18]: probably, yes

**Review by rdimitrov — APPROVED** [review]

**Comment by JAORMX** [comment]
superseded by https://github.com/stacklok/minder/pull/1937 (to reduce the noise from `protoc`)

File: pkg/api/protobuf/go/minder/v1/minder.pb.go (hunk 1 above)
Comment (AGMETEOR) [reply pkg/api/protobuf/go/minder/v1/minder.pb.go:18]: Is there a make target I can use to generate this file?

Comment (rdimitrov) [reply pkg/api/protobuf/go/minder/v1/minder.pb.go:18]: yeah, `make gen` takes care of all generation-related things 👍

**Pull request closed without merging**